tensorflow_io/core/kernels/parquet_kernels.cc

```diff
@@ -162,6 +162,9 @@ class ParquetReadableResource : public ResourceBase {
           row_group_reader->Column(column_index);
 
       // buffer to fill location is value.data()[row_to_read_start - start]
+      // Note: ReadBatch may not be able to read the elements requested
+      // (row_to_read_count) in one shot, as such we use while loop of
+      // `while (row_left > 0) {...}` to read until complete.
```

Comment:
@yongtang the puzzling thing was that this happened only for a particular column in the parquet dataset that was provided. Any reference as to why this might happen?

Reply:
Yes, it'd be great to understand how general this error is. Any workaround for now? When might this fix get into a release?

Reply:
@dgoldenberg-audiomack this PR should fix the issue. Also, can we use your sample parquet dataset in our CI tests (if that is fine with you)?

Reply:
@kvignesh1420 Use the parquet dataset - yes, you can absolutely. So the fix will be in the next release; do you know when that's slated for?

Reply:
@dgoldenberg-audiomack I am not sure about the next release, but you can always use `tensorflow-io-nightly` for using these immediate fixes.

Reply:
Will merge this after the tests pass.

Reply:
Thanks @kvignesh1420 @dgoldenberg-audiomack. The `ReadBatch` was only encountered for ByteArray. This is related to the internal implementation of Arrow's Parquet cpp. One discrepancy I can think of, is that ByteArray are different from other types where other types are more uniform with size known before hand (e.g., the size can be preallocated for float/bool/int as long as the number of rows are known). For ByteArray (variable size) the allocation might be hard to pre-allocate.